pkg/apis/apps/v1alpha1/imagepullnode_types.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2019 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// ImagePullNodeSpec defines the desired state of ImagePullNode
+type ImagePullNodeSpec struct {
+	// Specifies images to be pulled on this node
+	Images map[string]ImagePullSpec `json:"images,omitempty"`
+}
+
+// ImagePullSpec defines the pulling spec of an image
+type ImagePullSpec struct {
+	// ImagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling the image.
+	// If specified, these secrets will be passed to individual puller implementations for them to use.  For example,
+	// in the case of docker, only DockerConfig type secrets are honored.
+	// +optional
+	PullSecrets []types.NamespacedName `json:"pullSecrets,omitempty"`
+
+	// Tags is a list of versions of this image
+	Tags []ImageTagPullSpec `json:"tags"`
+}
+
+// ImageTagPullSpec defines the pulling spec of an image tag
+type ImageTagPullSpec struct {
+	// Specifies the image tag
+	Tag string `json:"tag"`
+
+	// Specifies the create time of this tag
+	// +optional
+	CreatedAt *metav1.Time `json:"createdAt,omitempty"`
+
+	// PullPolicy is an optional field to set parameters of the pulling task. If not specified,
+	// the system will use the default values.
+	// +optional
+	PullPolicy *ImageTagPullPolicy `json:"pullPolicy,omitempty"`
+
+	// List of objects depended by this object. If this image is managed by a controller,
+	// then an entry in this list will point to this controller.
+	// +optional
+	OwnerReferences []v1.ObjectReference `json:"ownerReferences,omitempty"`
+}
+
+// ImageTagPullPolicy defines the policy of the pulling task
+type ImageTagPullPolicy struct {
```

Comment:
Why is the ImageTagPullPolicy different from the PullPolicy in ImagePullJob? I assume they can be the same and share the same definition.

Reply:
Job and `Task` may still be some different attributes. I'm thinking of moving ImagePullNode to an internal package. This CR doesn't want to be handed over to users, only to use it internally.